setup.py

```diff
@@ -56,7 +60,7 @@
     )
     _ABLE_TO_COMPILE_EXTENSIONS = False
 
-if _ABLE_TO_COMPILE_EXTENSIONS:
+if _ABLE_TO_COMPILE_EXTENSIONS and not DISABLE_COMPILE_ARROW_EXTENSIONS:
```

Comment:
If you accept my other suggestion
```suggestion
if _ABLE_TO_COMPILE_EXTENSIONS and not SNOWFLAKE_DISABLE_COMPILE_ARROW_EXTENSIONS:
```